app/modules/users/permissions/rules.py

```diff
@@ -418,35 +423,49 @@ def _permitted_via_collaboration(self, user):
             (self._obj.__class__.__name__, self._action)
         )
 
-        for collab_assoc in user.get_collaboration_associations():
-            collab_users = collab_assoc.collaboration.get_users()
-            for other_user in collab_users:
-                if other_user not in tried_users:
-                    tried_users.append(other_user)
-
-                    if other_user.owns_object(self._obj):
-                        if (
-                            self._action == AccessOperation.READ
-                        ) & collab_assoc.collaboration.user_has_read_access(
-                            current_user.guid
-                        ):
-                            return True
-                        elif (
-                            self._action == AccessOperation.WRITE
-                        ) & collab_assoc.collaboration.user_has_edit_access(
-                            current_user.guid
-                        ):
-                            return True
-                        break
-
-                if object_user_methods is not None:
-                    for method in object_user_methods:
-                        if not hasattr(self._obj, method):
-                            log.warning(
-                                f'{self._obj.__class__.__name__} object does not have accessor {method}'
-                            )
-                        elif getattr(self._obj, method)(other_user):
-                            return True
+        if self._action == AccessOperation.READ:
```

Comment:
I don't see how this works, this is setting the approval_state to a class member variable without an object?